### PR TITLE
Trim leading locale segment in redirect url

### PR DESCRIPTION
**Pull request by mingwli**
- try trim leading locale when redirect document id

fixing [DevOps#141966](https://dev.azure.com/ceapex/Engineering/_workitems/edit/141966)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5307)

### CODE REVIEW

File: src/docfx/localization/LocalizationUtility.cs [hunk 1]
```diff
@@ -193,6 +193,29 @@ public static bool TryRemoveLocale(string name, out string nameWithoutLocale, ou
             return false;
         }
 
+        public static bool TryTrimPathLeadingLocale(PathString path, out PathString pathWithoutLocale, out string locale)
```
Comment (yufeih) [inline src/docfx/localization/LocalizationUtility.cs]:
```suggestion
        public static bool TryTrimPathLeadingLocale(string path, out string pathWithoutLocale, out string locale)
```

File: src/docfx/localization/LocalizationUtility.cs [hunk 2]
```diff
@@ -193,6 +193,29 @@ public static bool TryRemoveLocale(string name, out string nameWithoutLocale, ou
...
+        {
+            pathWithoutLocale = default;
+            locale = null;
+            if (string.IsNullOrEmpty(path))
+            {
+                return false;
+            }
+
+            var isAbsolute = path.Value[0] == '/';
```
Comment (yufeih) [inline src/docfx/localization/LocalizationUtility.cs]: when will path start with `/`?

Comment (mingwli) [reply src/docfx/localization/LocalizationUtility.cs]: Currently it won't, will limit this functionality inside redirection provider

File: src/docfx/build/redirection/RedirectionProvider.cs [hunk 3]
```diff
@@ -225,5 +221,24 @@ private static string RemoveTrailingIndex(string redirectionUrl)
...
+
+        private static string RemoveLeadingLocale(string redirectionUrl, string hostName)
+        {
+            var (redirectHostName, redirectPath) = UrlUtility.SplitBaseUrl(redirectionUrl);
+            if (!string.Equals(redirectHostName, hostName, StringComparison.OrdinalIgnoreCase))
+            {
+                return redirectionUrl;
+            }
+
+            if (redirectPath.IndexOf('/') is int slashIndex && slashIndex < 0)
```
Comment (live1206) [inline src/docfx/build/redirection/RedirectionProvider.cs]: isn't `redirectPath.IndexOf('/')` always `int`?

Comment (mingwli) [reply src/docfx/build/redirection/RedirectionProvider.cs]: 'slashIndex' is used later, and IndexOf doesn't have a 'out int index' overload.

**Pull request merged**